src/utilities/EventEmitter.test.ts

```diff
@@ -0,0 +1,242 @@
+      expect(eventEmitter.eventsOnce).toStrictEqual(Object.create(null));
+    });
+
+    it('Removes one listener from all lists', () => {
+      eventEmitter.off(listener);
+      expect(eventEmitter.any).toStrictEqual([altListener]);
+      expect(eventEmitter.anyOnce).toStrictEqual([altListener]);
+      expect(eventEmitter.events['myEvent']).not.toContain(listener);
+      expect(eventEmitter.eventsOnce['myEvent']).not.toContain(listener);
+    });
+
+    it('Removes a specific listener from one event', () => {
+      eventEmitter.off('myEvent', listener);
+      expect(eventEmitter.any).toStrictEqual([listener, altListener]);
+      expect(eventEmitter.anyOnce).toStrictEqual([listener, altListener]);
+      expect(eventEmitter.events['myEvent']).not.toContain(listener);
+      expect(eventEmitter.events['myOtherEvent']).toContain(listener);
+    });
+
+    it('Removes a specific listener from multiple events', () => {
+      eventEmitter.off(['myEvent', 'myOtherEvent'], listener);
+      expect(eventEmitter.events['myEvent']).not.toContain(listener);
+      expect(eventEmitter.events['myOtherEvent']).not.toContain(listener);
+      expect(eventEmitter.events['myThirdEvent']).toContain(listener);
+    });
+
+    it('Removes all listeners from an event', () => {
+      eventEmitter.off('myEvent');
+      expect(eventEmitter.events['myEvent']).toBe(undefined);
+      expect(eventEmitter.events['myOtherEvent']).toContain(listener);
+    });
+
+    it('Removes all listeners from multiple events', () => {
+      eventEmitter.off(['myEvent', 'myOtherEvent']);
+      expect(eventEmitter.events['myEvent']).toBe(undefined);
+      expect(eventEmitter.events['myOtherEvent']).toBe(undefined);
+      expect(eventEmitter.events['myThirdEvent']).toContain(listener);
+    });
+
+  });
+
+  describe('calling the listeners method', () => {
+    beforeEach(() => {
+      eventEmitter = new EventEmitter();
+      eventEmitter.on(listener);
+      eventEmitter.on(altListener);
+      eventEmitter.on('myEvent', listener);
+      eventEmitter.once(listener);
+      eventEmitter.once(altListener);
+      eventEmitter.once('myEvent', listener);
+      eventEmitter.once('myEvent', altListener);
+    });
+
+    it('Returns all listeners for a given event', () => {
+      expect(eventEmitter.listeners('myEvent')).toHaveLength(3);
+      expect(eventEmitter.listeners('myOtherEvent')).toBe(null);
+      eventEmitter.on('myEvent', listener);
+      expect(eventEmitter.listeners('myEvent')).toHaveLength(4);
+      eventEmitter.once('myEvent', listener);
+      expect(eventEmitter.listeners('myEvent')).toHaveLength(5);
+    });
+
+  });
+
+  describe('calling the once method', () => {
+    beforeAll(() => {
+      eventEmitter = new EventEmitter();
+    });
+
+    it('Adds a listener to anyOnce on calling `once` with a listener', () => {
+      eventEmitter.once(listener);
+      expect(eventEmitter.anyOnce).toHaveLength(1);
```

Comment:
As per the above comments, I'd expect us here to use `emit` twice and check if the `listener` was called only once.